tools/ramdump/dumpParser.py

```diff
@@ -834,6 +834,165 @@ def main():
 
+		# Power Management Parser
+		fd = open(config_path, 'r')
+		data = fd.read()
+		fd.close()
+
+		if not 'CONFIG_PM_METRICS=y' in data:
+			print('PM_METRICS is not enable. Enable PM_METRICS to see power management information')
+			return
+
+		g_pmglobals = rParser.get_address_of_symbol("g_pmglobals")
+
+		PM_STATUS = ("PM_NORMAL", "PM_IDLE", "PM_STANDBY", "PM_SLEEP")
+
+		# This information depends on the g_pmglobals structure
+		MAX_SIZE_MEMBER = 4
+		QUOTIENT = 0
+
+		SIZE_OF_REGSEM = 0
+		if 'CONFIG_BINMGR_RECOVERY=y' in data:
+			SIZE_OF_REGSEM += 4
+
+		SIZE_OF_REGSEM += 4
+		if 'CONFIG_PRIORITY_INHERITANCE=y' or 'CONFIG_BINMGR_RECOVERY=y' in data:
+			if 'CONFIG_SEM_PREALLOCHOLDERS=' in data:
+				index = data.find('CONFIG_SEM_PREALLOCHOLDERS=')
+				index += len('CONFIG_SEM_PREALLOCHOLDERS=')
+				PREALLOCHOLDERS = 0
+				while data[index] != '\n':
+					PREALLOCHOLDERS *= 10
+					PREALLOCHOLDERS += int(data[index])
+					index += 1
+				if PREALLOCHOLDERS == 0:
+					SIZE_OF_REGSEM += 8
+				else:
+					SIZE_OF_REGSEM += 4
+
+		REGISTRY_POINT = SIZE_OF_REGSEM
+		SIZE_OF_REGISTRY = 8
+
+		if 'CONFIG_SYSTEM_TIME64=y' in data:
+			MAX_SIZE_MEMBER = 8
+
+		NUMBER_OF_DOMAINS = 0
+		if 'CONFIG_PM_NDOMAINS=' in data:
+			index = data.find('CONFIG_PM_NDOMAINS=')
+			index += len('CONFIG_PM_NDOMAINS=')
+			while data[index] != '\n':
+				NUMBER_OF_DOMAINS *= 10
+				NUMBER_OF_DOMAINS += int(data[index])
+				index += 1
+		
+		NUMBER_OF_PM_MEMORY = 0
+		if 'CONFIG_PM_MEMORY=' in data:
+			index = data.find('CONFIG_PM_MEMORY=')
+			index += len('CONFIG_PM_MEMORY=')
+			while data[index] != '\n':
+				NUMBER_OF_PM_MEMORY *= 10
+				NUMBER_OF_PM_MEMORY += int(data[index])
+				index += 1
+
+		DOMAIN_POINT = REGISTRY_POINT + SIZE_OF_REGISTRY
+		for i in range(0, NUMBER_OF_DOMAINS):
+			if (DOMAIN_POINT % MAX_SIZE_MEMBER) != 0:
+				QUOTIENT = (int)(DOMAIN_POINT / MAX_SIZE_MEMBER) + 1
+				DOMAIN_POINT = QUOTIENT * MAX_SIZE_MEMBER
+
+			SIZE_OF_DOMAIN = 6
+
+			if NUMBER_OF_PM_MEMORY > 1:
+				SIZE_OF_DOMAIN += (2 * (NUMBER_OF_PM_MEMORY - 1))
+
+			MAX_SIZE_MEMBER = 4
+			if (SIZE_OF_DOMAIN % MAX_SIZE_MEMBER) != 0:
+				QUOTIENT = (int)(SIZE_OF_DOMAIN / MAX_SIZE_MEMBER) + 1
+				SIZE_OF_DOMAIN = QUOTIENT * MAX_SIZE_MEMBER
+
+			start_history = rParser.read_word(g_pmglobals + DOMAIN_POINT + SIZE_OF_DOMAIN)
+			end_history = rParser.read_word(g_pmglobals + DOMAIN_POINT + SIZE_OF_DOMAIN + 4)
+
+			STATE_STAY_POINT = g_pmglobals + DOMAIN_POINT + SIZE_OF_DOMAIN + 20
+			print('')
+			print('Domain number is %d'%i)
+			print('***************************')
+			print('Stay time for each PM state')
+			print('***************************')
+			print('')
+
+			print('************************')
+			print('   State  |  Stay(sec) |')
+			print('----------|------------|')
+			for i in range(0, 4):
+				STAY_TIME = rParser.read_halfword(STATE_STAY_POINT + (i * 2))
+				print('{:<10}|'.format(PM_STATUS[i]), '{:^11}|'.format(STAY_TIME))
```

Comment:
How about this?
```suggestion
				print(f'{PM_STATUS[i]:<10}| {STAY_TIME:^11}|')
```

Reply:
Thank you for your comments : )
But it makes build error in my enviroment.
Is this right syntax ??

Reply:
Which version of python do you use?
If below code work on your environment, how about this?
```suggestion
				print('{:<10}|{:^11}|'.format(PM_STATUS[i], STAY_TIME))
```

Reply:
It works well : )
Thank you.